kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala

```diff
@@ -117,16 +116,16 @@ object Logging {
     // This distinguishes the log4j 1.2 binding, currently
     // org.slf4j.impl.Log4jLoggerFactory, from the log4j 2.0 binding, currently
     // org.apache.logging.slf4j.Log4jLoggerFactory
-    val binderClass = StaticLoggerBinder.getSingleton.getLoggerFactoryClassStr
-    "org.slf4j.impl.Log4jLoggerFactory".equals(binderClass)
+    "org.slf4j.impl.Log4jLoggerFactory"
+      .equals(LoggerFactory.getILoggerFactory.getClass.getName)
```

Comment:
to adapt slf4j-2.x, Spark bumped the slf4j to version 2.0.3 in [SPARK-40511](https://github.com/apache/spark/pull/37844).

```
  org.apache.kyuubi.KyuubiSQLException: org.apache.kyuubi.KyuubiSQLException: Exception in thread "main" java.lang.NoClassDefFoundError: org/slf4j/impl/StaticLoggerBinder
	at org.apache.kyuubi.Logging$.isLog4j2(Logging.scala:128)
	at org.apache.kyuubi.Logging$.org$apache$kyuubi$Logging$$initializeLogging(Logging.scala:158)
	at org.apache.kyuubi.Logging.initializeLoggerIfNecessary(Logging.scala:103)
	at org.apache.kyuubi.Logging.initializeLoggerIfNecessary$(Logging.scala:99)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine$.initializeLoggerIfNecessary(SparkSQLEngine.scala:152)
	at org.apache.kyuubi.Logging.logger(Logging.scala:45)
	at org.apache.kyuubi.Logging.logger$(Logging.scala:43)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine$.logger(SparkSQLEngine.scala:152)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine$.<init>(SparkSQLEngine.scala:168)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine$.<clinit>(SparkSQLEngine.scala)
	at org.apache.kyuubi.engine.spark.SparkSQLEngine.main(SparkSQLEngine.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:1009)
	at org.apache.spark.deploy.SparkSubmit$$anon$1.run(SparkSubmit.scala:166)
	at org.apache.spark.deploy.SparkSubmit$$anon$1.run(SparkSubmit.scala:164)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1878)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:164)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:204)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1100)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1109)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.ClassNotFoundException: org.slf4j.impl.StaticLoggerBinder
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
 See more: /home/runner/work/kyuubi/kyuubi/kyuubi-server/target/work/runner/kyuubi-spark-sql-engine.log.0
	at org.apache.kyuubi.KyuubiSQLException$.apply(KyuubiSQLException.scala:69)
	at org.apache.kyuubi.engine.ProcBuilder.$anonfun$start$1(ProcBuilder.scala:229)
	at java.lang.Thread.run(Thread.java:750)
```

Reply:
we can make a separating PR for master/1.7 if it's a valid fix

Reply:
will do.

Reply:
https://github.com/apache/kyuubi/pull/4407